src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2004,7 +2005,34 @@ bool entitiesQuery
   auto_ptr<DBClientCursor>  cursor;
   Query                     query(finalQuery.obj());
 
-  query.sort(BSON(ENT_CREATION_DATE << 1));
+  if (sorted == "")
+  {
+    query.sort(BSON(ENT_CREATION_DATE << 1));
+  }
+  else
+  {
+    std::vector<std::string>  sortedV;
+    int components = stringSplit(sorted, ',', sortedV);
+    BSONObjBuilder sortOrder;
+    for (int ix = 0; ix < components; ix++)
+    {
+      std::string sortToken;
+      int sortDirection;
+      if (sortedV[ix][0] == '!')
+      {
+        // reverse
+        sortToken = sortedV[ix].substr(1);
+        sortDirection = -1;
+      }
+      else
+      {
+        sortToken = sortedV[ix];
+        sortDirection = 1;
+      }
+      sortOrder.append(std::string(ENT_ATTRS) + "." + sortToken + "." + ENT_ATTRS_VALUE, sortDirection);
+    }
+    query.sort(sortOrder.obj());
+  }
 
```

Comment:
Wow, not one single empty line here ...
I see just a mass of characters, might only be me having these problems ... :-)

Style guide:
* Indent local variables, and add an empty lines after the local var declarations.
* After blocks it is a good idea to add an empty line (before sortOrder.append(... ) and query.sort(sortOrder...)
* Indent '='s

This might just be my opinion, but I really think the code gets easier to understand like this ...